benchmark/clBench/clblas/blas2/trsv.cpp

```diff
@@ -0,0 +1,239 @@
+    *success = false;
+    return;
+  }
+
+  err = clEnqueueWriteBuffer(executorPtr->queue(), m_a_gpu, CL_TRUE, 0,
+                             lda * n * sizeof(scalar_t), m_a.data(), 0, nullptr,
+                             nullptr);
+  err = clEnqueueWriteBuffer(executorPtr->queue(), v_x_gpu, CL_TRUE, 0,
+                             xlen * sizeof(scalar_t), v_x.data(), 0, nullptr,
+                             nullptr);
+
+  if (err != CL_SUCCESS) {
+    state.SkipWithError("error writing opencl buffers");
+    *success = false;
+    return;
+  }
+
+  cl_command_queue clQueue = executorPtr->queue();
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> v_x_ref = v_x;
+  reference_blas::trsv(uplo_str, t_str, diag_str, n, m_a.data(), lda,
+                       v_x_ref.data(), incX);
+  std::vector<scalar_t> v_x_temp = v_x;
+  {
+    err = clEnqueueWriteBuffer(executorPtr->queue(), v_x_temp_gpu, CL_TRUE, 0,
+                               xlen * sizeof(scalar_t), v_x_temp.data(), 0,
+                               nullptr, nullptr);
+
+    cl_event event;
+    err = clblasStrsv(layout, a_uplo, a_tr, a_diag, n, m_a_gpu, 0, lda,
+                      v_x_temp_gpu, 0, incX, 1, &clQueue, 0, nullptr, &event);
+    err = clWaitForEvents(1, &event);
+
+    if (err != CL_SUCCESS) {
+      state.SkipWithError("error running clblasStrsm");
+      *success = false;
+      return;
+    }
+
+    err = clEnqueueReadBuffer(clQueue, v_x_temp_gpu, CL_TRUE, 0,
+                              xlen * sizeof(scalar_t), v_x_temp.data(), 0,
+                              nullptr, nullptr);
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(v_x_temp, v_x_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_method_def = [&]() -> std::vector<cl_event> {
+    cl_event event;
+    clblasStatus ret =
+        clblasStrsv(layout, a_uplo, a_tr, a_diag, n, m_a_gpu, 0, lda,
+                    v_x_temp_gpu, 0, incX, 1, &clQueue, 0, nullptr, &event);
+
+    if (ret != clblasSuccess) {
+      *success = false;
+      state.SkipWithError("Failed");
+      return {};
+    } else {
+      CLEventHandler::wait(event);
+      return {event};
+    }
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  blas_benchmark::utils::calc_avg_counters(state);
+}
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args, ExecutorType* exPtr,
+                        bool* success) {
+  auto tbmv_params = blas_benchmark::utils::get_tbmv_params(args);
```

Comment:
This should be `trsv_params` here on the left hand side.

Reply:
I think we should use the name of the operator being implemented here (`trsv` instead of `tbmv`) on the left hand side. This would make the name of the variable consistent with the operator being implemented. 
On a side note, I know that most of the benchmarks in SYCL-BLAS use `gemm_params` for the name of this variable, which is definitely something we will need to fix somewhere down the line, but for now I think it's best if we keep the param name as that of the operator being implemented.